src/service/QuestionGenerator.js

```diff
@@ -0,0 +1,50 @@
+import { randomNumberInRange } from './randomNumberInRange';
+import {QuestionService} from './QuestionService';
+import { WHO_IS_THAT_POKEMON } from "./modes";
+
+const questionService = new QuestionService()
+
+export class QuestionGenerator {
+
+    constructor(mode) {
+        this.mode = mode;
+        this.minPokeId = 1;
+        this.maxPokeId = 152;
+        this.askedQuestions = [];
+        this.askedQuestionsCount = 0;
+    };
+
+    async getNextQuestion() {
+        if (this.askedQuestionsCount === 30) {
+            return undefined;
+        }
+        this.askedQuestionsCount++;
```

Comment:
@lukaszdutka Incrementor doesn't work. When I run getNextQuestion method it is still equal to 0. Do you know what is wrong?

Reply:
This test works for me:
```
it('Should increment counter', async () => {
    //given
    const startingCount = generatedQuestion.askedQuestionsCount;

    //when
    await generatedQuestion.getNextQuestion();

    //then
    expect(generatedQuestion.askedQuestionsCount).toEqual(startingCount+1);
  });
  ```

Reply:
It is good idea to test it in this way. I added it to test file.